tests/browser/test/toolbox_drag_test.js

```diff
@@ -0,0 +1,202 @@
+  'Images',
+  'Emoji! o((*^ᴗ^*))o',
+  'Validators',
+  'Mutators',
+  'Style',
+  'Serialization',
+];
+
+/**
+ * Check whether an element is inside the bounds of the Blockly div. You can use this
+ * to determine whether a block on the workspace or flyout is inside the Blockly div.
+ * This does not check whether there are other Blockly elements (such as a toolbox or
+ * flyout) on top of the element.
+ * @param browser The active WebdriverIO Browser object.
+ * @param element The element to look for.
+ * @returns A Promise resolving to true if the element is in bounds and false otherwise.
+ */
+async function elementInBounds(browser, element) {
+  return await browser.execute((elem) => {
+    const rect = elem.getBoundingClientRect();
+
+    const blocklyDiv = document.getElementById('blocklyDiv');
+    const blocklyRect = blocklyDiv.getBoundingClientRect();
+
+    const vertInView =
+      rect.top >= blocklyRect.top && rect.bottom <= blocklyRect.bottom;
+    const horInView =
+      rect.left >= blocklyRect.left && rect.right <= blocklyRect.right;
+
+    return vertInView && horInView;
+  }, element);
+}
+
+/**
+ * Get how many top-level blocks there are in the specified category.
+ * @param browser The active WebdriverIO Browser object.
+ * @param categoryName The name of the category to inspect.
+ * @returns A Promise resolving to the number of top-level blocks in the specified
+ *     category's flyout.
+ */
+async function getBlockCount(browser, categoryName) {
+  const category = await getCategory(browser, categoryName);
+  await category.click();
+  await browser.pause(PAUSE_TIME);
+
+  const blockCount = await browser.execute(() => {
+    return Blockly.getMainWorkspace()
+      .getFlyout()
+      .getWorkspace()
+      .getTopBlocks(false).length;
+  });
+
+  // Unicode escape to close flyout.
+  await browser.keys(['\uE00C']);
+  await browser.pause(PAUSE_TIME);
+  return blockCount;
+}
+
+async function isBlockDisabled(browser, i) {
+  return await browser.execute((n) => {
+    return !Blockly.getMainWorkspace()
+      .getFlyout()
+      .getWorkspace()
+      .getTopBlocks()
+      [n].isEnabled();
+  }, i);
+}
+
+/**
+ * Loop over a list of categories and click on each one to open it.
+ * @param browser The WebdriverIO Browser instance for this test.
+ * @param categoryList An array of category names, as strings.
+ * @returns A Promise that resolves when all actions have finished.
+ */
+async function openCategories(browser, categoryList) {
+  let failureCount = 0;
+  for (const categoryName of categoryList) {
+    const blockCount = await getBlockCount(browser, categoryName);
+
+    try {
+      for (let i = 0; i < blockCount; i++) {
+        const category = await getCategory(browser, categoryName);
+        await category.click();
+        if (await isBlockDisabled(browser, i)) {
+          // Unicode escape to close flyout.
+          await browser.keys(['\uE00C']);
+          await browser.pause(PAUSE_TIME);
```

Comment:
Can you use a `continue` here instead of wrapping the rest of the method in an `else`?

Reply:
I like `else` more than `continue` :)

Reply:
I think for long term code health it's better to keep conditional blocks short: https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

For smaller things I can see how `else` is clearer. But in this case (especially since it is already wrapped in a `try...catch`) I would really prefer to de-dent it.